### PR TITLE
Allow parameter scaling for stable convergence

### PR DESCRIPTION
**Pull request by avik-pal**
Ref https://github.com/SciML/GalacticOptim.jl/issues/146

### CODE REVIEW

File: src/solve.jl [hunk 1]
```diff
@@ -1,4 +1,34 @@
...
+               scale::Bool = false,
+               scaling_functions::Union{Nothing, Tuple{Function, Function}} = nothing,
+               cb = (args...) -> (false),
+               kwargs...)
+    !scale && return __solve(prob, opt, args...; kwargs...)
+
+    θ_start = copy(prob.u0)
+    
+    scale_function, inv_scale_function =
+        isnothing(scaling_functions) ? (x -> x ./ θ_start, x -> x .* θ_start) : scaling_functions
```
Comment (ChrisRackauckas) [inline src/solve.jl:12]: Make the scaling function a callable struct to cut down on recompile times?

**Review by ChrisRackauckas — APPROVED** [review]
This looks good?

**Comment by avik-pal** [comment]
Yeah. I will open a downsteam test PR in GalacticOptim

File: src/solve.jl [hunk 2]
```diff
@@ -1,4 +1,46 @@
...
+    compute_inverse ? x .* inv_scale.scale : x ./ inv_scale.scale
+
+function solve(prob::OptimizationProblem, opt, args...;
+               scale::Bool = false, scaling_function = nothing,
+               cb = (args...) -> (false), kwargs...)
+    !scale && return __solve(prob, opt, args...; kwargs...)
+
+    θ_start = copy(prob.u0)
+
+    if isnothing(scaling_function) && any(iszero.(θ_start))
```
Comment (YingboMa) [inline src/solve.jl:16]: Never use `isnothing`

**Comment by Vaibhavdixit02** [comment]
Why was this closed?

**Pull request closed without merging**